modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java

```diff
@@ -221,8 +230,17 @@ public void validate(
 					getObjectValidationRuleEngine(
 						objectValidationRule.getEngine());
 
-			Map<String, Object> results = objectValidationRuleEngine.execute(
-				variables, objectValidationRule.getScript());
+			if (StringUtil.equals(
+					objectValidationRuleEngine.getName(),
+					ObjectValidationRuleConstants.ENGINE_TYPE_DDM)) {
+
+				results = objectValidationRuleEngine.execute(
+					inputObjects, objectValidationRule.getScript());
+			}
+			else {
+				results = objectValidationRuleEngine.execute(
+					variables, objectValidationRule.getScript());
+			}
```

Comment:
@p-albuquerque why do we have two different maps now? `inputObjects` and `variables` it's not intuitive. What is the difference between them?